File: CalculadoraBasica/C++/Calculadora.cpp
/*
 * Calculadora
 * crea una calculadora con interfaz utlizando las minimas lineas de codigo posibles
 * que sea modular
 * EXTRA: Gestion de errores
 */
#include <iostream>
using namespace std;

//Funciones
//Funcion para sumar
float sumar(float num1, float num2)
{
    return num1 + num2;
}
//Funcion para restar
float restar(float num1, float num2)
{
    return num1 - num2;
}
//Funcion para multiplicar
float multiplicar(float num1, float num2)
{
    return num1 * num2;
}
//Funcion para dividir
float dividir(float num1, float num2)
{
    //Gestion de errores
    if ((num1 == 0) || (num2 == 0))
    {
        /* error */
        throw runtime_error("Error de division por cero");
    }
    else
    {
        return num1 / num2;
    }
}

int main(int argc, char const *argv[])
{
    //Variables
    int respuesta = 0;
    int resultado = 0;
    float num1 = 0;
    float num2 = 0;

    //Gestion de errores
    try
    {
        cout << "|-----------------------|" << endl;
        cout << "|0. Salir.              |" << endl;
        cout << "|1. Sumar.              |" << endl;
        cout << "|2. Restar.             |" << endl;
        cout << "|3. Multiplicar.        |" << endl;
        cout << "|4. Dividir.            |" << endl;
        cout << "|-----------------------|" << endl;
        cout << "Elige una opcion del menu: " << endl;
        cin >> respuesta;

        do
        {
            /* code */
            switch (respuesta)
            {
            case 0:
                /* Salir */
                cout << "El programa va a finalizar." << endl;
                break;
            case 1:
                /* Sumar */
                cout << "Has elegido la opcion de sumar." << endl;
                cout << "Introduce el primer numero: " << endl;
                cin >> num1;
                cout << "Introduce el segundo numero: " << endl;
                cin >> num2;
                //Almacenamos el resultado e imprimimos
                resultado = sumar(num1, num2);
                cout << "El resultado de sumar los numeros " << num1 << " y " << num2 << " es: " << resultado << endl;
                break;
            case 2:
                /* Restar */
                cout << "Has elegido la opcion de restar." << endl;
                cout << "Introduce el primer numero: " << endl;
                cin >> num1;
                cout << "Introduce el segundo numero: " << endl;
                cin >> num2;
                //Almacenamos el resultado e imprimimos
                resultado = restar(num1, num2);
                cout << "El resultado de restar los numeros " << num1 << " y " << num2 << " es: " << resultado << endl;
                break;
            case 3:
                /* Multiplicar */
                cout << "Has elegido la opcion de multiplicar." << endl;
                cout << "Introduce el primer numero: " << endl;
                cin >> num1;
                cout << "Introduce el segundo numero: " << endl;
                cin >> num2;
                //Almacenamos el resultado e imprimimos
                resultado = multiplicar(num1, num2);
                cout << "El resultado de multiplicar los numeros " << num1 << " y " << num2 << " es: " << resultado << endl;
                break;
            case 4:
                /* Dividir */
                cout << "Has elegido la opcion de dividir." << endl;
                cout << "Introduce el primer numero: " << endl;
                cin >> num1;
                cout << "Introduce el segundo numero: " << endl;
                cin >> num2;
                //Almacenamos el resultado e imprimimos
                resultado = dividir(num1, num2);
                cout << "El resultado de dividir los numeros " << num1 << " y " << num2 << " es: " << resultado << endl;
                break;
            default:
                break;
            }
        } while ((respuesta > 0) && (respuesta < 5));
    }
    catch(const std::exception& e)
    {
        cout << "Ha ocurrido un error y el programa va a finalizar." << endl;
    }
 return 0;
}

File: CalculadoraBasica/Python/Calculadora.py
# Calculadora
# crea una calculadora con interfaz utlizando las minimas lineas de codigo posibles
# que sea modular
# EXTRA: Gestion de errores
#Funciones
#Funcion para sumar
def sumar(num_1, num_2):
    return num_1 + num_2

#Funcion para restar
def restar(num_1, num_2):
    return num_1 - num_2

#Funcion para multiplicar
def multiplicar(num_1, num_2):
    return num_1 * num_2

#Funcion para dividir
def dividir(num_1, num_2):
    if num_2 == 0:
        return "Error de division por cero"
    else:
        return num_1 / num_2

# Preguntamos al usuario qué función quiere realizar
print("|-----------------------|")
print("|0. Salir.              |")
print("|1. Sumar.              |")
print("|2. Restar.             |")
print("|3. Multiplicar.        |")
print("|4. Dividir.            |")
print("|-----------------------|")
respuesta = int(input("Elige una opcion del menu:")) #Almacenamos la respuesta en una variable

# Creamos las condiciones
if respuesta == 1:
    #Sumar
    print("Has elegido la opcion de sumar.")
    num_1 = float(input("Introduce el primer numero: "))
    num_2 = float(input("Introduce el segundo numero: "))
    # Almacenamos el resultado y lo mostramos por pantalla
    resultado = sumar(num_1, num_2)
    print(f"El resultado de sumar los numeros {num_1} y {num_2} es: {resultado}")
elif respuesta == 2:
    #Restar
    print("Has elegido la opcion de restar.")
    num_1 = float(input("Introduce el primer numero: "))
    num_2 = float(input("Introduce el segundo numero: "))
    # Almacenamos el resultado y lo mostramos por pantalla
    resultado = restar(num_1, num_2)
    print(f"El resultado de restar los numeros {num_1} y {num_2} es: {resultado}")
elif respuesta == 3:
    #Multiplicar
    print("Has elegido la opcion de multiplicar.")
    num_1 = float(input("Introduce el primer numero: "))
    num_2 = float(input("Introduce el segundo numero: "))
    # Almacenamos el resultado y lo mostramos por pantalla
    resultado = multiplicar(num_1, num_2)
    print(f"El resultado de multiplicar los numeros {num_1} y {num_2} es: {resultado}")
elif respuesta == 4:
    #Dividir
    print("Has elegido la opcion de dividir.")
    num_1 = float(input("Introduce el primer numero: "))
    num_2 = float(input("Introduce el segundo numero: "))
    # Almacenamos el resultado y lo mostramos por pantalla
    resultado = dividir(num_1, num_2)
    print(f"El resultado de dividir los numeros {num_1} y {num_2} es: {resultado}")
elif respuesta == 0:
    #Salir
    print("El programa ha finalizado.")
else:
    print("Has introducido un numero que no está en la lista o algún caracter.")
